### ISSUE TITLE
Problem with authentication

### ISSUE THREAD

**Issue by cmacetko** [firebase/firebase-js-sdk#3081]
Hello people

Next, I noticed a problem with the browser opened by Google on android, it raises the exception:
`Access to XMLHttpRequest at 'https://www.googleapis.com/identitytoolkit/v3/relyingparty/getAccountInfo?key=XXXXX from origin 'https://www.XXXXXXX.com.br' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

My js script:
```
firebase.auth().signInWithEmailAndPassword("XXX", "XX").then(function() {			
}).catch(function(error) {
});
```

I noticed this problem only in the browser opened by the google assistant

**Comment by berndartmueller** [comment]
Maybe same or related to the issue I'm experiencing with a lot of our users: https://github.com/firebase/firebase-js-sdk/issues/2652#issuecomment-630209037

Can you always reproduce it?

> Our users experience the same `auth/network-request-failed` firebase auth issues since some time now. It mainly happens to Android Chrome users and the issue start immediately after/during using `firebase.auth().signInAnonymously`. (Note: I'm not using react, I simply use plain old Node.js SSR with vanilla js)
> 
> I'm also having a hard time reproducing the issue, I tried many real devices as well as devices on Browserstack. I'm just not getting this issue.
> 
> I already made sure that
> 
> * there is no `form` html tag used
> * the call to action button is a `<button type='button'>Signup</button>` element
> * no session cookie/token revocation happens
> 
> Any ideas or any idea how I could reproduce it myself? Thanks a lot!

**Comment by cmacetko** [comment]
Hello

It is similar, in my case I am authenticating the user with an email and password previously created and I can simulate using the "google assistant" opening the site for him.

**Comment by berndartmueller** [comment]
Maybe an explanation can be found here ->  https://support.google.com/chrome/thread/11089651?hl=en 

But no clue how to fix it for affected users

**Comment by Dragoncall** [comment]
Hello there! We have the same issue on mobile Chrome browsers. Some of our users receive a network exception. This also happens in the Samsung browser. I have not been able to reproduce this personally, but a large cohort of our users is afflicted.

**Comment by berndartmueller** [comment]
While using Sentry to log errors I was able to spot the first time this error occurred. Interestingly, the following XHR request was sent shortly before the network timeout error was thrown by firebase: 

`POST https://me.kis.v2.scr.kaspersky-labs.com/8F4A3112-AF04-F245-BDB1-07C58C08DDD4/DAD682E4-7B8A-9546-98AB-5CB26B97741B/to/ajaxRequestNotify`

And as firebase returns a http status code of `0` for the failed auth request, there are a few assumptions that can be made from it:

- Some kind of antivirus/proxy/firewall blocks requests
- XHR request gets canceled because navigation happened
- CORS issues

I have now started to show a feedback popup to those affected users to try to get in touch with them and debug it in greater detail. However, if it's really caused by some kind of external firewall, then the last resort would be to stop using Firebase SDK on the client and shift firebase authentication to our Node.js server via Firebase REST.

**Comment by cmacetko** [comment]
Hello

I believe that the problem is not the firewall, I managed to simulate the problem right here by stopping the firewalls on my network.
There seems to be some bug on the google side.

* I started watching reports of this problem about 1 week ago
* Attached is information obtained from the sentry and from a remote debug in the chrome browser
![2020-05-20_13h22_52](https://user-images.githubusercontent.com/1896228/82472765-0e61f600-9a9f-11ea-8fbc-7b34ad4a4dae.png)
![2020-05-20_13h23_09](https://user-images.githubusercontent.com/1896228/82472767-0efa8c80-9a9f-11ea-95d5-df0b61a85776.png)
![2020-05-20_13h23_15](https://user-images.githubusercontent.com/1896228/82472770-0f932300-9a9f-11ea-9aaa-a44e81710809.png)
![2020-05-20_13h30_47](https://user-images.githubusercontent.com/1896228/82472771-0f932300-9a9f-11ea-8728-a52a89cfb536.png)
![2020-05-20_13h30_59](https://user-images.githubusercontent.com/1896228/82472774-102bb980-9a9f-11ea-8f1a-e521d85c50b7.png)
![2020-05-20_13h31_13](https://user-images.githubusercontent.com/1896228/82472778-102bb980-9a9f-11ea-9802-71961ef3654f.png)

Here is the script used for the test:
```
<script src="https://www.gstatic.com/firebasejs/7.14.4/firebase-app.js"></script>
<script src="https://www.gstatic.com/firebasejs/7.14.4/firebase-database.js"></script>
<script src="https://www.gstatic.com/firebasejs/7.14.4/firebase-auth.js"></script>

<script language="javascript" type="text/javascript">
try 
{

	var config 				= {
	apiKey: "[Hide]",
	databaseURL: "[Hide]"
	};
	firebase.initializeApp(config);

	firebase.auth().signInWithEmailAndPassword("[Hide]", "[Hide]").then(function() {

		console.log("Ok");

	}).catch(function(error) {

		console.log(error);

	});

} catch (e) {
	
	console.log(e);
	
}
</script>
```

**Comment by berndartmueller** [comment]
@cmacetko I have exactly the same errors in Sentry. 

On your screenshot it clearly shows that it's a CORS issue.

Are you testing on an actual mobile device? Could you please outline the steps needed to reproduce, maybe I'm lucky to be able to also simulate the issue.

Yesterday I also contacted the official Firebase support regarding this issue, but still waiting for their response.

**Comment by cmacetko** [comment]
@berndartmueller I'm testing it on a real phone, then I'll make a video showing the problem, and how to simulate it, and attach it here

**Comment by berndartmueller** [comment]
@cmacetko Have you been able to reproduce it and make a video of it? I got a reply from firebase support and they would need a reproducible demo project for further help

**Comment by cmacetko** [comment]
@berndartmueller It seems that it was solved, I carried out new tests without presenting the problem in the same conditions as before.
I noticed that now the google server is returning a valid access-control-allow-origin.

**Comment by cmacetko** [comment]
I will continue to monitor but it seems that it has been resolved.

**Comment by berndartmueller** [comment]
@cmacetko I'm experiencing the same. This issue has not happened to our users since 2 days, as before it used to happen to 10% of our Android Chrome users per day. 

Don't know what happened, but I'm just glad it's fixed.

If it occurs again, we can open a new issue again.

Thanks a lot!

**Comment by avolkovi** [comment]
Closing out for now since it seems that the issue is resolved, please re-open if you see this happening again

**Comment by MAlecuPenta** [comment]
It's not solved at all guys - there is no way to replace `<form>` with `<div>` in firebaseui. Check this [post](https://github.com/firebase/firebaseui-web/issues/261#issuecomment-662618889).
